en/sensor/px4flow.md

```diff
@@ -0,0 +1,456 @@
+* USB bootloader
+* USB serial up to 921600 baud (including live camera view with [QGroundControl](http://qgroundcontrol.com/))
+* USB power option
+* **Does fit the MatrixVision Bluefox MV mounting holes (camera center off-centered)**
+
+![PX4Flow Top](../../assets/hardware/sensors/px4flow/px4flow_top.jpg) ![px4flow-bottom](../../assets/hardware/sensors/px4flow/px4flow_bottom.jpg) 
+
+
+# Pixhawk Setup
+
+> **Note** Applies to all PX4 FMU versions, including all versions of Pixhawk.
+
+- Update the firmware on PX4Flow using *QGroundControl* (in the top left menu, click on CONFIG, then on Firmware Upgrade)
+- Connect PX4Flow I2C to the Pixhawk I2C
+
+The module will be detected on boot, flow data should be coming through at 10Hz if the autopilot is connected via **USB**. 
+Flow data is transmitted via wireless at a lower rate.
+
+## Orientation
+
+The default orientation (meaning: zero rotation) is defined as Y on flow board pointing towards **front of vehicle** as shown in the following picture.
+
+![PX4Flow align with Pixhawk](../../assets/hardware/sensors/px4flow/px4flowalignwithpixhawk.jpg)
+
+
+The orientation of PX4Flow can be set using the parameter [MAV_SENSOR_ORIENTATION](https://mavlink.io/en/messages/common.html#MAV_SENSOR_ORIENTATION)
+
+> **Warning** This parameter can only be applied to PX4 flight stack and is **not** part of the sensor itself settings
+
+<!-- Remove below if orientation  param above is correct -->
+<!-- 
+The detail parameter is listed below and more can be found [here](???). 
+| **SENS_FLOW_ROT** | PX4Flow board rotation |   |   |  0 |
+| ::: | <div>This parameter defines the rotation of the PX4FLOW board relative to the platform.
+Zero rotation is defined as Y on flow board pointing towards front of vehicle
+Possible values are:
+0 = No rotation
+1 = Yaw 45°
+2 = Yaw 90°
+3 = Yaw 135°
+4 = Yaw 180°
+5 = Yaw 225°
+6 = Yaw 270°
+7 = Yaw 315°</div> ||||
+-->
+
+## Wiring
+
+It is highly recommended to use twisted / shielded I2C wires to minimize radiated noise ([I2C spec](http://www.nxp.com/documents/user_manual/UM10204.pdf)). From the spec:
+
+If the bus lines are twisted-pairs, each bus line must be twisted with a VSS return. Alternatively, the SCL line can be twisted with a VSS return, and the SDA line twisted with a VDD return. In the latter case, capacitors must be used to decouple the VDD line to the VSS line at both ends of the twisted pairs.
+If the bus lines are shielded (shield connected to VSS), interference is minimized. However, the shielded cable must have low capacitive coupling between the SDA and SCL lines to minimize crosstalk.
+
+
+
+## I2C
+
+The 7 Bit **I2C Address** of the Flow Module is user selectable. Using the solder jumpers on the back of the flow board you can add an offset to the baseaddress 0x42. The address range for the 8 possible choices is: 0x42 - 0x49.
```

Comment:
I think most of this section actually belongs in devguide right? My understanding is that all I2C thingies are connected in the same way - if not, can we get instructions on how you would connect it to pixhawk?

Reply:
Right, they are connected the same way. Each device has an address and if you have multiple devices on the same bus, all addresses need to be different. This is why the px4flow module allows to change the address (which happens typically via solder jumpers), and it's important to mention it.
It's one of those things I don't know if it belongs to the dev- or user-guide. A user connecting different devices to a single I2C bus should know about addresses.

Reply:
@bkueng Thank you. So how about this fix. 
1. We remove all the stuff immediately under Wiring (because any connection to I2C is the same - according to above). Specifically, take away this bit:
   > It is highly recommended to use twisted / shielded I2C wires to minimize radiated noise ([I2C spec](http://www.nxp.com/documents/user_manual/UM10204.pdf)). From the spec:
   >
   > If the bus lines are twisted-pairs, each bus line must be twisted with a VSS return. Alternatively, the SCL line can be twisted with a VSS return, and the SDA line twisted with a VDD return. In the latter case, capacitors must be used to decouple the VDD line to the VSS line at both ends of the twisted pairs.
   > If the bus lines are shielded (shield connected to VSS), interference is minimized. However, the shielded cable must have low capacitive coupling between the SDA and SCL lines to minimize crosstalk.
2. Instead we add a statement. 
   > The PX4flow is connected to the I2C bus. If you are connecting multiple devices to the same bus you will need to set each with a unique address. The next section explains how.
3. We rename this section from `## I2C` to `Set PX4FLOW I2C Device address`
4. What/who is the bit below :`The **I2C frame** contains 22 bytes of data in the following order:`. Doesn't look like is part of setting the I2C address. I am guessing it is driver implementation information. What do you think. It isn't generic or I would suggest having it in http://dev.px4.io/en/sensor_bus/i2c.html .

Reply:
1-3: sounds good
4: It describes the communication protocol and should IMO be documented in the driver: https://github.com/PX4/Firmware/blob/master/src/drivers/px4flow/i2c_frame.h. I can move it there if you wish.

Reply:
Thanks Beat! I'll do 1-3 tomorrow, and delete "4". If you could move the comms part of the protocol that would be much appreciated.

Reply:
Fixed in https://github.com/PX4/px4_user_guide/pull/286/commits/7e89e5265be6dc02868266ee096d38e97ec811f8

Thank @bkueng , merging now.